symfony/plugins/orangehrmCoreOAuthPlugin/Api/OAuthClientAPI.php

```diff
@@ -0,0 +1,221 @@
+
+
+class OAuthClientAPI extends Endpoint implements CrudEndpoint
+{
+
+    public const PARAMETER_CLIENT_ID = 'clientId';
+    public const PARAMETER_CLIENT_SECRET = 'clientSecret';
+    public const PARAMETER_REDIRECT_URI = 'redirectUri';
+
+    /**
+     * @var null|OAuthService
+     */
+    protected ?OAuthService $oAuthService = null;
+
+
+    /**
+     * @return OAuthService
+     */
+    public function getOAuthService(): OAuthService
+    {
+        if (is_null($this->oAuthService)) {
+            $this->oAuthService = new OAuthService();
+        }
+        return $this->oAuthService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $oAuthClientSearchFilterParams = new OAuthClientSearchFilterParams();
+        $this->setSortingAndPaginationParams($oAuthClientSearchFilterParams);
+
+        $count = $this->getOAuthService()->getOAuthClientDao()->getOAuthClientsCount($oAuthClientSearchFilterParams);
+
+        $oauthClients = $this->getOAuthService()->getOAuthClientDao()->getOAuthClients($oAuthClientSearchFilterParams);
+        return new EndpointCollectionResult(
+            OAuthClientModel::class,
+            $oauthClients,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            ...$this->getSortingAndPaginationParamsRules(OAuthClientSearchFilterParams::ALLOWED_SORT_FIELDS)
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
+    {
+        // TODO:: Check data group permission
+        $oAuthClient = $this->saveOAuthClient();
+
+        return new EndpointResourceResult(OAuthClientModel::class, $oAuthClient);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            ...$this->getCommonBodyValidationRules(),
+        );
+    }
+
+    /**
+     * @return ParamRule[]
+     */
+    private function getCommonBodyValidationRules(): array
+    {
+        return [
+            new ParamRule(self::PARAMETER_CLIENT_ID),
+            new ParamRule(self::PARAMETER_CLIENT_SECRET),
+            new ParamRule(self::PARAMETER_REDIRECT_URI),
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function delete(): EndpointResult
+    {
+        $ids = $this->getRequestParams()->getArray(RequestParams::PARAM_TYPE_BODY, CommonParams::PARAMETER_IDS);
+        $this->getOAuthService()->deleteOAuthClients($ids);
+        return new EndpointResourceResult(ArrayModel::class, $ids);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForDelete(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(CommonParams::PARAMETER_IDS),
```

Comment:
better to handle undeletable ids in BE as well.
Can refer 

https://github.com/orangehrm/orangehrm/blob/5.0/symfony/plugins/orangehrmAdminPlugin/Api/UserAPI.php#L246-L263